robot/impl/resource_manager.go

```diff
@@ -514,11 +515,12 @@ func (manager *resourceManager) Close(ctx context.Context) error {
 	return allErrs
 }
 
-// completeConfig process the tree in reverse order and attempts to build
-// or reconfigure resources that are wrapped in a placeholderResource.
+// completeConfig process the tree in reverse order and attempts to build or reconfigure
+// resources that are wrapped in a placeholderResource.
 func (manager *resourceManager) completeConfig(
 	ctx context.Context,
 	lr *localRobot,
+	forceSync bool,
 ) {
```

Comment:
@dgottlieb heads up, I changed the flow of the function quite a bit since you last reviewed (and it ended up removing most our discussion threads from the code diff :sob:) - I'm hoping [this comment](https://github.com/viamrobotics/rdk/pull/3707#discussion_r1604131495) summarized the change.

More than happy to discuss IRL if the change stream is hard to follow.